include/revng/StackAnalysis/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,645 @@
+      // model::Register::xmm0_x86,
+      // model::Register::xmm1_x86,
+      // model::Register::xmm2_x86
+    };
+  static constexpr std::array<model::Register::Values, 0>
+    VectorReturnValueRegisters = {
+      // model::Register::st0_x86,
+      // model::Register::xmm0_x86,
+    };
+
+  static constexpr std::array CalleeSavedRegisters = {
+    model::Register::ebx_x86,
+    model::Register::ebp_x86,
+    model::Register::esp_x86,
+    model::Register::edi_x86,
+    model::Register::esi_x86
+  };
+};
+
+/// `regparm(1)` SystemV x86 ABI
+/// Function arguments are passed using both stack and specified registers.
+template<>
+struct CallingConventionTrait<model::abi::SystemV_x86_regparm_1> {
+  static constexpr bool ArgumentsArePositionBased = false;
+  static constexpr bool AllowPassingAggregatesInRegisters = false;
+  static constexpr size_t GenericRegistersUsedPerAggregateArgument = 1;
+  static constexpr size_t GenericRegistersUsedPerAggregateReturnValue = 2;
+  static constexpr bool CalleeIsResponsibleForStackCleanup = false;
+  static constexpr size_t StackAlignment = 16;
+
+  static constexpr std::array GenericArgumentRegisters = {
+    model::Register::eax_x86
+  };
+  static constexpr std::array GenericReturnValueRegisters = {
+    model::Register::eax_x86,
+    model::Register::edx_x86
+  };
+
+  static constexpr std::array<model::Register::Values, 0>
+    VectorArgumentRegisters = {
+      // model::Register::xmm0_x86,
+      // model::Register::xmm1_x86,
+      // model::Register::xmm2_x86
+    };
+  static constexpr std::array<model::Register::Values, 0>
+    VectorReturnValueRegisters = {
+      // model::Register::st0_x86,
+      // model::Register::xmm0_x86,
+    };
+
+  static constexpr std::array CalleeSavedRegisters = {
+    model::Register::ebx_x86,
+    model::Register::ebp_x86,
+    model::Register::esp_x86,
+    model::Register::edi_x86,
+    model::Register::esi_x86
+  };
+};
+
+/// `regparm(2)` SystemV x86 ABI
+/// Function arguments are passed using both stack and specified registers.
+template<>
+struct CallingConventionTrait<model::abi::SystemV_x86_regparm_2> {
+  static constexpr bool ArgumentsArePositionBased = false;
+  static constexpr bool AllowPassingAggregatesInRegisters = false;
+  static constexpr size_t GenericRegistersUsedPerAggregateArgument = 1;
+  static constexpr size_t GenericRegistersUsedPerAggregateReturnValue = 2;
+  static constexpr bool CalleeIsResponsibleForStackCleanup = false;
+  static constexpr size_t StackAlignment = 16;
+
+  static constexpr std::array GenericArgumentRegisters = {
+    model::Register::eax_x86,
+    model::Register::edx_x86
+  };
+  static constexpr std::array GenericReturnValueRegisters = {
+    model::Register::eax_x86,
+    model::Register::edx_x86
+  };
+
+  static constexpr std::array<model::Register::Values, 0>
+    VectorArgumentRegisters = {
+      // model::Register::xmm0_x86,
+      // model::Register::xmm1_x86,
+      // model::Register::xmm2_x86
```

Comment:
I'm thinking, let's just add those to the model.
Even if we currently SA (StackAnalysis) doesn't emit them, it doesn't matter.
The ABI has its own dignity independently from other parts of the pipeline.

tl;dr: feel free to add any registers. Just make they don't alias other registers.